File: output/llama_4_maverick/avatar/algo_based_translation/java/python/successful_translation/codeforces_567_A.py

def main():
    n = int(input())
    arr = list(map(int, input().split()))
    for i in range(n):
        max_diff = max(abs(arr[i] - arr[0]), abs(arr[-1] - arr[i]))
        if i == 0:
            min_diff = arr[i + 1] - arr[i]
        elif i == n - 1:
            min_diff = arr[i] - arr[i - 1]
        else:
            min_diff = min(arr[i] - arr[i - 1], arr[i + 1] - arr[i])
        print(min_diff, max_diff)

if __name__ == "__main__":
    main()

File: output/deepseek_r1/avatar/algo_based_translation/python/java/codeforces_567_A.java

import java.util.Scanner;

public class codeforces_567_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int inp = scanner.nextInt();
        scanner.nextLine();
        String[] inputArray = scanner.nextLine().split(" ");
        int[] lista = new int[inputArray.length];
        for (int i = 0; i < inputArray.length; i++) {
            lista[i] = Integer.parseInt(inputArray[i]);
        }
        int n = lista.length;
        for (int i = 0; i < n; i++) {
            if (i == 0) {
                int minDiff = lista[i + 1] - lista[i];
                int maxDiff = lista[n - 1] - lista[i];
                System.out.println(minDiff + " " + maxDiff);
            } else if (i == n - 1) {
                int minDiff = lista[i] - lista[i - 1];
                int maxDiff = lista[i] - lista[0];
                System.out.println(minDiff + " " + maxDiff);
            } else {
                int prevDiff = lista[i] - lista[i - 1];
                int nextDiff = lista[i + 1] - lista[i];
                if (prevDiff > nextDiff) {
                    System.out.print(nextDiff + " ");
                } else {
                    System.out.print(prevDiff + " ");
                }
                int fromFirst = lista[i] - lista[0];
                int fromLast = lista[n - 1] - lista[i];
                System.out.println(Math.max(fromFirst, fromLast));
            }
        }
        scanner.close();
    }
}

